### PR TITLE
OBSDOCS-1405: Add docs for the Count Connector

### PR DESCRIPTION
**Pull request by max-cx**
:warning: DO NOT MERGE! `do-not-merge/hold`

:warning: This PR is tied to a product release date, which is currently set to **December 10**.

- If the merge review gets delayed, you're welcome to ping `mleonov` on Slack.
- If you have any questions, ping `mleonov` on Slack.

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16, 4.17, 4.18

Issue:

https://issues.redhat.com/browse/OBSDOCS-1405

Link to docs preview:

https://84281--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-collector/otel-collector-connectors.html#count-connector_otel-collector-connectors

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (iblancasa) [reply observability/otel/otel-collector/otel-collector-connectors.adoc]: Yes

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (kowen-rh) [inline observability/otel/otel-collector/otel-collector-connectors.adoc:14]: I believe "and" is a typo here: "pipelines and using"

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (kowen-rh) [inline observability/otel/otel-collector/otel-collector-connectors.adoc:100]: Is this actually a procedure or just a list of things to check? Must the checks be done in this order? How do we perform these checks?

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (kowen-rh) [inline observability/otel/otel-collector/otel-collector-connectors.adoc:112]: Do not use Latin abbreviations such as `e.g.` according to the IBM Style Guide. Use "for example".

**Review by kowen-rh — COMMENTED** [review]
Left a few comments, but overall this looks good to me!

/remove-label peer-review-in-progress
/label peer-review-done

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (IshwarKanse) [inline observability/otel/otel-collector/otel-collector-connectors.adoc:16]: The metric names are self explanatory. I leave it to the docs team to see if we should add it as a table like the upstream [docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/connector/countconnector#default-configuration  ).

**Review by IshwarKanse — APPROVED** [review]

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:100]: @kowen-rh, my apologies. I haven't edited this section yet.

Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:100]: @iblancasa, cc @kowen-rh, I'd like to incorporate all of this into into the connector's description for two reasons:
1. It's better that we provide this info initially than have the customer fail and then look for troubleshooting help.
2. We don't have troubleshooting sections for the rest of the OTEL components.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (iblancasa) [inline observability/otel/otel-collector/otel-collector-connectors.adoc:19]: I think this would be easier for customers
```suggestion
The following are the default exposed metrics:
```

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (iblancasa) [inline observability/otel/otel-collector/otel-collector-connectors.adoc:25]:
```suggestion
* `trace.span.count`
* `trace.span.event.count`
* `metric.count`
* `metric.datapoint.count`
* `log.record.count`
```

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (iblancasa) [inline observability/otel/otel-collector/otel-collector-connectors.adoc:31]: These metrics are tha names you set in the configuration. I think we can remove this

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:19]: Oops. Thanks for spotting this. Fixed.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc]: TY. Fixed.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (iblancasa) [inline observability/otel/otel-collector/otel-collector-connectors.adoc:27]:
```suggestion
You can also expose custom metric names.
```

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc]: TY. Fixed.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:27]: TY. Done.

**Review by iblancasa — APPROVED** [review]

**Review by xenolinux — COMMENTED** [review]
Added some suggestions and a few questions. Otherwise good job!!

I will remove the merge-review-in-progress label for now. Once you make updates and PR is ready to be merged, please add the merge-review-needed label again.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (iblancasa) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:109]: `<custom_metric_description>` was a placeholder let's say. We can place there whatever we want. Like "Example description for the metric"

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (iblancasa) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:83]: I think `<1> In this example, the exposed metric counts spans with the specified conditions.` is correct because the metric gives the number of spans where the conditions are true.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (iblancasa) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:58]: > In <1>'s description, it's not clear if grpc is a name of the receiver or a type of the receiver or a protocol that communicates with the receiver. Can you please check with SME/QE to clarify?
`grpc` is the protocol used by the receiver

> Similar comment for <2>'s description. Can users use exporters other than Prometheus, are they supported?
Yes. They can use others. The original change was correct.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:109]: TY. Updated.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:87]: TY.
Updated the first sentence.
Left the second sentence untouched (the suggestion would change the meaning).

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:83]: Thank you, but I won't apply this suggestion.
The first sentence is correct.
The suggested change for the second sentence would change the intended meaning.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:19]: Thank you, but I'd rather avoid phrasing this as a procedure.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc:58]: Thank you, both.
I deleted the callouts `<1>` and `<2>` because they are self-explanatory.
I also shortened the callouts `<4>` and `<5>` even more than suggested.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc]: Thank you for your feedback. I'm addressing this in the other thread that you started above.

File: observability/otel/otel-collector/otel-collector-connectors.adoc
Comment (max-cx) [reply observability/otel/otel-collector/otel-collector-connectors.adoc]: @iblancasa, WDYT? I'm proposing an edit here: https://github.com/openshift/openshift-docs/pull/84281/commits/e8068ee7e5e4aed4775143658d88c82e687c2691.

Comment (iblancasa) [reply observability/otel/otel-collector/otel-collector-connectors.adoc]: LGTM

**Review by skrthomas — COMMENTED** [review]
Merge review looks great. Nice work!

**Pull request merged**
